packages/store/ts/config/parseStoreConfig.ts

```diff
@@ -170,45 +200,58 @@ export const zEnumsConfig = z.object({
  *
  ************************************************************************/
+// Manual *UserConfig is used instead of `z.input<typeof zStoreConfig>` because zod can't preserve doc comments
 
-// zod doesn't preserve doc comments
-export type StoreUserConfig<
-  EnumNames extends StringForUnion = StringForUnion,
-  StaticUserTypes extends ExtractUserTypes<EnumNames> = ExtractUserTypes<EnumNames>
-> = EnumsConfig<EnumNames> & {
+/** Options without generics */
+export interface StoreSimpleOptions {
   /** The namespace for table ids. Default is "" (empty string) */
   namespace?: string;
   /** Path for store package imports. Default is "@latticexyz/store/src/" */
   storeImportPath?: string;
-  /**
-   * Configuration for each table.
-   *
-   * The key is the table name (capitalized).
-   *
-   * The value:
-   *  - abi or user type for a single-value table.
-   *  - FullTableConfig object for multi-value tables (or for customizable options).
-   */
-  tables: TablesConfig<AsDependent<StaticUserTypes>, AsDependent<StaticUserTypes>>;
   /** Path to the file where common user types will be generated and imported from. Default is "Types" */
   userTypesPath?: string;
   /** Path to the directory where generated files will be placed. (Default is "codegen") */
   codegenDirectory?: string;
-};
+}
+
+/** MUDCoreUserConfig wrapper to use generics in some options for better type inference */
+export type MUDUserConfig<
+  EnumNames extends StringForUnion = StringForUnion,
+  StaticUserTypes extends ExtractUserTypes<EnumNames> = ExtractUserTypes<EnumNames>
+> = MUDCoreUserConfig &
+  EnumsConfig<EnumNames> & {
+    /**
+     * Configuration for each table.
+     *
+     * The key is the table name (capitalized).
+     *
+     * The value:
+     *  - abi or user type for a single-value table.
+     *  - FullTableConfig object for multi-value tables (or for customizable options).
+     */
+    tables: TablesConfig<AsDependent<StaticUserTypes>, AsDependent<StaticUserTypes>>;
+  };
+
+export type MUDConfig<
+  EnumNames extends StringForUnion = StringForUnion,
+  StaticUserTypes extends ExtractUserTypes<EnumNames> = ExtractUserTypes<EnumNames>
+> = MUDCoreConfig &
+  FullEnumsConfig<EnumNames> & {
+    tables: FullTablesConfig<StaticUserTypes, StaticUserTypes>;
+  };
 
-/** Type helper for defining StoreUserConfig */
-export function storeConfig<
+/** mudCoreConfig wrapper to use generics in some options for better type inference */
+export function mudConfig<
   // (`never` is overridden by inference, so only the defined enums can be used by default)
   EnumNames extends StringForUnion = never,
   StaticUserTypes extends ExtractUserTypes<EnumNames> = ExtractUserTypes<EnumNames>
->(config: StoreUserConfig<EnumNames, StaticUserTypes>) {
-  return config;
+>(config: MUDUserConfig<EnumNames, StaticUserTypes>): MUDConfig<EnumNames, StaticUserTypes> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return mudCoreConfig(config) as any;
```

Comment:
for context - why do we have to return this as any?

Reply:
because zod doesn't have plugins (https://github.com/latticexyz/mud/pull/679#discussion_r1181621049), and for `parseWorldConfig` it doesn't have any of the store stuff either